### ISSUE TITLE
Openshift-pipelines tekton-pipelines-webook pods stuck in terminating state

### ISSUE THREAD

**Issue by HumairAK** [operate-first/operations#297]
https://console-openshift-console.apps.zero.massopen.cloud/k8s/ns/openshift-pipelines/pods

This is resulting in a flood of errors on the openshift-console alerting. 

```
~ $ oc get pods -n openshift-pipelines --field-selector status.phase=Pending
NAME                                        READY   STATUS        RESTARTS   AGE
tekton-pipelines-webhook-5ff9f7c9f7-28lm5   0/1     Terminating   0          23h
tekton-pipelines-webhook-5ff9f7c9f7-2czgh   0/1     Terminating   0          4d11h
tekton-pipelines-webhook-5ff9f7c9f7-2r5m9   0/1     Terminating   0          4d9h
tekton-pipelines-webhook-5ff9f7c9f7-4n5mn   0/1     Terminating   0          4d12h
tekton-pipelines-webhook-5ff9f7c9f7-5pw9z   0/1     Terminating   0          24h
tekton-pipelines-webhook-5ff9f7c9f7-898px   0/1     Terminating   0          21h
tekton-pipelines-webhook-5ff9f7c9f7-ghdxd   0/1     Terminating   0          4d9h
tekton-pipelines-webhook-5ff9f7c9f7-lmmgk   0/1     Terminating   0          21h
tekton-pipelines-webhook-5ff9f7c9f7-lw7vl   0/1     Terminating   0          4d21h
tekton-pipelines-webhook-5ff9f7c9f7-psnmf   0/1     Terminating   0          4d6h
tekton-pipelines-webhook-5ff9f7c9f7-rqmcr   0/1     Terminating   0          34h
tekton-pipelines-webhook-5ff9f7c9f7-vx66t   0/1     Terminating   0          4d21h
~ $ 

```

**Comment by HumairAK** [comment]
Seems like other namespaces have terminating stuck pods as well: 

```bash
~ $ oc get pods --field-selector status.phase=Pending -A | awk '{if($4=="Terminating"){print $1 "    \t" $2}}'
ds-ml-workflows-ws    	jupyterhub-9010-deploy
istio-system    	cluster-local-gateway-5898bc5c74-xs5ft
lars-sandbox    	update-zones-periodic-1620235800-7xbgh
lars-sandbox    	update-zones-periodic-1620324900-jr455
lars-sandbox    	update-zones-periodic-1620338400-g8fwx
lars-sandbox    	update-zones-periodic-1620404100-vd72c
lars-sandbox    	update-zones-periodic-1620425700-5n7ql
lars-sandbox    	update-zones-periodic-1620440100-gbsms
lars-sandbox    	update-zones-periodic-1620451800-2wst6
lars-sandbox    	update-zones-periodic-1620648000-xw8x6
lars-sandbox    	update-zones-periodic-1620649800-k957n
lars-sandbox    	update-zones-periodic-1620675900-qwwbf
lars-sandbox    	update-zones-periodic-1620676800-xtbnr
lars-sandbox    	update-zones-periodic-1620679500-ttpvh
lars-sandbox    	update-zones-periodic-1620721800-7qhpb
lars-sandbox    	update-zones-periodic-1620722700-zdxs5
openshift-pipelines    	tekton-pipelines-webhook-5ff9f7c9f7-28lm5
openshift-pipelines    	tekton-pipelines-webhook-5ff9f7c9f7-2czgh
openshift-pipelines    	tekton-pipelines-webhook-5ff9f7c9f7-2r5m9
openshift-pipelines    	tekton-pipelines-webhook-5ff9f7c9f7-4n5mn
openshift-pipelines    	tekton-pipelines-webhook-5ff9f7c9f7-5pw9z
openshift-pipelines    	tekton-pipelines-webhook-5ff9f7c9f7-898px
openshift-pipelines    	tekton-pipelines-webhook-5ff9f7c9f7-dprxl
openshift-pipelines    	tekton-pipelines-webhook-5ff9f7c9f7-gbgb4
openshift-pipelines    	tekton-pipelines-webhook-5ff9f7c9f7-ghdxd
openshift-pipelines    	tekton-pipelines-webhook-5ff9f7c9f7-lmmgk
openshift-pipelines    	tekton-pipelines-webhook-5ff9f7c9f7-lw7vl
openshift-pipelines    	tekton-pipelines-webhook-5ff9f7c9f7-psnmf
openshift-pipelines    	tekton-pipelines-webhook-5ff9f7c9f7-rqmcr
openshift-pipelines    	tekton-pipelines-webhook-5ff9f7c9f7-vx66t
openshift-pipelines    	tekton-pipelines-webhook-5ff9f7c9f7-wpblk
opf-ci-pipelines    	aicoe-pipelinerun-2045f26c-67e2-4675-8b75-bb2bf87c9c0c-st-klk5j
opf-ci-pipelines    	aicoe-pipelinerun-3d4f068d-69c4-4d09-ae77-65066b5cd174-st-lxxqc
opf-ci-pipelines    	aicoe-pipelinerun-d79ab790-c557-4e19-b6a3-fe6776ec1adc-pr-csf8m
opf-ci-pipelines    	aicoe-pipelinerun-d79ab790-c557-4e19-b6a3-fe6776ec1adc-py-drdt4
opf-ci-pipelines    	aicoe-pipelinerun-f7947bff-0747-45d8-bbc1-da246ccac9b5-st-gj6fm
~ $ 


```

**Comment by tumido** [comment]
Pods seems to be stuck in terminating, because the container is still pending creation:

```
status:
  phase: Pending
  conditions:
    - type: Initialized
      status: 'True'
      lastProbeTime: null
      lastTransitionTime: '2021-05-07T06:48:44Z'
    - type: Ready
      status: 'False'
      lastProbeTime: null
      lastTransitionTime: '2021-05-07T06:48:44Z'
      reason: ContainersNotReady
      message: 'containers with unready status: [webhook]'
    - type: ContainersReady
      status: 'False'
      lastProbeTime: null
      lastTransitionTime: '2021-05-07T06:48:44Z'
      reason: ContainersNotReady
      message: 'containers with unready status: [webhook]'
    - type: PodScheduled
      status: 'True'
      lastProbeTime: null
      lastTransitionTime: '2021-05-07T06:48:44Z'
  hostIP: 192.12.185.132
  startTime: '2021-05-07T06:48:44Z'
  containerStatuses:
    - name: webhook
      state:
        waiting:
          reason: ContainerCreating
      lastState: {}
      ready: false
      restartCount: 0
      image: >-
        registry.redhat.io/openshift-pipelines/pipelines-webhook-rhel8@sha256:be16516aea9cd4a632a1f15d62be421f4da816472f862f8813fe65bd2053abff
      imageID: ''
      started: false
  qosClass: Burstable
```

Seems like some variation to: https://access.redhat.com/solutions/4627811
I'll ask around on openshift forums.

**Comment by tumido** [comment]
Openshift SDN forum seems to be unsure if this can be caused by image pull failures. But we can't provide events for that anymore. Will update here if the investigation leads anywhere new.

**Comment by tumido** [comment]
Openshift team seems to stop paying attention to us after I've told them we don't have events for the time this occured. I think we should just try deleting the Terminating pods right now and be on alert if it happens any time again.

**Comment by HumairAK** [comment]
I cleared the terminating pods, which brought the number of alerts down from 41 to 10

**Comment by tumido** [comment]
It seems we have 2 new pods stuck in terminating state there today. Events don't seem to offer any explanation.

I suggest to keep deleting those pods manually for now and once we clean up everything with a Kubeflow upgrade we should also clean up and reinstall openshift pipelines operator to verify this is not caused by some obsolete resource left from Kubeflow or something.

![image](https://user-images.githubusercontent.com/7453394/118488279-dfb34580-b71b-11eb-9b2a-4b4e17842e37.png)

All pods are owned from the same replicaset. But there seems to be other replicasets in the namespace scaled to 0 pods which suggests to me there's something wrong with the setup...

**Comment by HumairAK** [comment]
bumping,

similar issue now in `smaug` cluster: 

![image](https://user-images.githubusercontent.com/10904967/133118477-8c88e268-11b1-4bb6-9895-c8cdb5007dd6.png)

**Comment by HumairAK** [comment]
Okay for some reason we're seeing 3 pods get created simulatneously, then get terminated instantly: 

![image](https://user-images.githubusercontent.com/10904967/133124483-6d343a2c-4699-4fd8-a5d9-7d4b179d068b.png)

**Comment by HumairAK** [comment]
For some reason the deployment is fluctuating from 4 to 1 

![image](https://user-images.githubusercontent.com/10904967/133124886-216076fd-4ec3-42a3-9609-9d71b38802a8.png)

**Comment by HumairAK** [comment]
I changed the `HorizontalPodAutoscaler` named `tekton-pipelines-webhook` to: 
```yaml
spec:
  scaleTargetRef:
    kind: Deployment
    name: tekton-pipelines-webhook
    apiVersion: apps/v1
  minReplicas: 1
  maxReplicas: 1
  targetCPUUtilizationPercentage: 100
```

From `maxReplicas: 5`. It says This resources is managed by `TektonPipeline` but does not seem like the change is being reverted. For now this seems like it's relaxed the constant autoscaling up/down of the deployment. I still don't know what is causing it though.

**Comment by HumairAK** [comment]
@vdemeester has kindly offered to help investigate this issue on the `smaug` cluster, we have provided him with the appropriate access needed to help dig into it and find the underlying cause.

**Comment by HumairAK** [comment]
Upstream issue created by @vdemeester : https://github.com/tektoncd/operator/issues/491
RH Issue: https://issues.redhat.com/browse/SRVKP-1783

**Comment by vdemeester** [comment]
@HumairAK yes this has been fixed upstream and should be available as part of the latest 1.6 release of OpenShift Pipelines (1.6.2 in a few days but the fix should have been included in 1.6.0 already)

**Comment by HumairAK** [comment]
resolved after ocp upgrade (and thus ocp pipelines update to 1.6.2
